clang/include/clang/CIR/Dialect/IR/CIROps.td

```diff
@@ -3165,6 +3165,42 @@ def TrapOp : CIR_Op<"trap", [Terminator]> {
   let assemblyFormat = "attr-dict";
 }
 
+//===----------------------------------------------------------------------===//
+// PrefetchOp
+//===----------------------------------------------------------------------===//
+
+def CIR_PrefetchOp : CIR_Op<"prefetch"> {
+  let summary = "prefetch operation";
+  let description = [{
+    The `cir.prefetch` op prefetches data from the given address.
+
+    ```mlir
+    cir.prefetch(%0 : !cir.ptr<!void>, true, 1, true)
+    ```
+
+    This opcode has the three attributes:
+    1. The $is_write is the specifier determining if the fetch should be for a
+    read or write.
+    2. The $locality is a temporal locality specifier
+    ranging from (0) - no locality, to (3) - extremely local keep in cache.
+    3. The $is_data_cache type specifies whether the prefetch is performed on
+    the data or instruction cache.
+  }];
+
+  let arguments = (
+    ins CIR_PointerType:$addr,
+        BoolAttr:$is_write,
+        ConfinedAttr<I32Attr, [IntMinValue<0>,
+          IntMaxValue<3>]>:$locality,
+        BoolAttr:$is_data_cache);
```

Comment:
This last one seems to be LLVM IR specific and isn't really part of the builtin, you can leave it out and make sure you pass "true" directly when lowering to LLVM.